### PR TITLE
[Merged by Bors] - TO-3294 instrument user api

### PR DESCRIPTION
**Pull request by finiteprods**
**Summary**

adds instrumentation to the main web service so requests can be traced through the service.

### CODE REVIEW

File: discovery_engine_core/web-api/src/handlers.rs
Comment (biernacki) [inline discovery_engine_core/web-api/src/handlers.rs:39]: are we interested also how much time ranking takes? @acrrd

File: discovery_engine_core/web-api/src/handlers.rs
Comment (biernacki) [inline discovery_engine_core/web-api/src/handlers.rs:69]: maybe a `debug` in else block with the `user_id` and `body.document_id` that was not found if that doesn't show up already in log output from `warp::trace::request()`

**Review by biernacki — COMMENTED** [review]
lgtm, just a couple of questions

**Review by biernacki — APPROVED** [review]
lgtm, let me know if you'll need another review depending on @acrrd response

File: discovery_engine_core/web-api/src/handlers.rs
Comment (acrrd) [reply discovery_engine_core/web-api/src/handlers.rs:39]: In the whole request. We can do it here but maybe we can do it in a more comprehensive way from the load balancer.
we can avoid it for a moment and understand later what to do.

**Comment by finiteprods** [comment]
bors merge

**Comment by finiteprods** [comment]
bors merge

**Comment by finiteprods** [comment]
bors merge

**Pull request closed without merging**
